### PR TITLE
Imagebox: add proper svg resizing

### PR DESCRIPTION
**Pull request by Crylia**
This is part one of a small fix for imagebox. This PR is so imagebox correctly resized SVG's, previously it was just rendered by cairo (which worked fine) but for the svg to be properly recolored and rendered the Rsvg object needs to know about its future size, thats why I had to switch to render_document (its what the docs recommend to use anyway) which now takes an extra RsvgRect argument which is a simple stuct with its w/h and x/y(not needed) information

I also added seperate logic to scale the cairo surface. This was needed because the original scale would also automatically scale the SVG *again*, instead It now scales the rect, and the temporary surface (I needed a new one with the new size, else the original wouldn't fit the resized SVG anymore)

### CODE REVIEW

**Comment by Crylia** [comment]
*facepalm* I'll fix the formatting thing (thanks github addon for randomly opening the PR lol)

File: lib/wibox/widget/imagebox.lua [hunk 1]
```diff
@@ -182,6 +182,9 @@ function imagebox:draw(ctx, cr, width, height)
 
     local w, h = self._private.default.width, self._private.default.height
 
+    local RsvgRect = Rsvg.Rectangle {x = 0, y = 0, width = w, height = h}
```
Comment (actionless) [inline lib/wibox/widget/imagebox.lua:185]: what if image is not an svg though?

Comment (Crylia) [reply lib/wibox/widget/imagebox.lua:185]: Right, I could simply move it to where I apply it and just `or` it. I'll do it some time this week together with part 2 for the recoloring container

Comment (actionless) [reply lib/wibox/widget/imagebox.lua:185]: my point was that it seems that one of unit tests related to images is failing when invoked with non-svg image:

https://github.com/awesomeWM/awesome/actions/runs/4834063772/jobs/8614833487?pr=3802

File: lib/wibox/widget/imagebox.lua [hunk 2]
```diff
@@ -233,7 +236,12 @@ function imagebox:draw(ctx, cr, width, height)
             cr:clip(self._private.clip_shape(cr, w*aspects.w, h*aspects.h, unpack(self._private.clip_args)))
         end
 
-        cr:scale(aspects.w, aspects.h)
+        if self._private.handle then
+            RsvgRect = Rsvg.Rectangle {x = 0, y = 0, width = w*aspects.w, height = h*aspects.h }
+            svg_surf = cairo.ImageSurface(cairo.Format.ARGB32, w*aspects.w, h*aspects.h)
```
Comment (actionless) [inline lib/wibox/widget/imagebox.lua:241]: the stacktrace from the failed test actually pointing to this line, but anyway you got the idea :)

File: lib/wibox/widget/imagebox.lua [hunk 3]
```diff
@@ -253,10 +260,16 @@ function imagebox:draw(ctx, cr, width, height)
...
+        if self._private.handle then
+            -- Translation is not needed in the new surface
+            svg_surf = cairo.ImageSurface(cairo.Format.ARGB32, 0, 0)
+            self._private.handle:render_document(cr, Rsvg.Rectangle {x = 0, y = 0, width = w, height = h})
+        end
     end
 
     if self._private.handle then
-        self._private.handle:render_cairo(cr)
+        cr:set_source_surface(svg_surf, 0, 0)
```
Comment (sclu1034) [inline lib/wibox/widget/imagebox.lua:272]: This surface has never been drawn to, so what's the point of setting it as a source?

**Review by sclu1034 — CHANGES_REQUESTED** [review]

**Review by Elv13 — APPROVED** [review]

**Comment by Elv13** [comment]
@Crylia apology for the long delay. I wasn't paying attention much to all PRs lately. Can you rebase this so the CI can run?

**Comment by actionless** [comment]
@Elv13  i've re-opened it to re-run - it should merge without manual rebase

File: lib/wibox/widget/imagebox.lua (hunk 2 above)
Comment (actionless) [reply lib/wibox/widget/imagebox.lua:239]: you marked thread as resolved but test is still failing on this line: https://github.com/awesomeWM/awesome/actions/runs/6042238743/job/16477140356?pr=3802

**Review by actionless — CHANGES_REQUESTED** [review]

**Comment by actionless** [comment]
also aside from failing test, code-style check is also not passing: https://github.com/awesomeWM/awesome/actions/runs/6042238743/job/16477141387?pr=3802

File: lib/wibox/widget/imagebox.lua (hunk 3 above)
Comment (Elv13) [reply lib/wibox/widget/imagebox.lua:272]: Done in https://github.com/awesomeWM/awesome/pull/3881

**Comment by Elv13** [comment]
Moved to https://github.com/awesomeWM/awesome/pull/3881 because I don't have write access to this PR

**Pull request closed without merging**